kiwmi/input/cursor.c

```diff
@@ -307,3 +270,62 @@ cursor_destroy(struct kiwmi_cursor *cursor)
 
     free(cursor);
 }
+
+void
+cursor_refresh_focus(
+    struct kiwmi_cursor *cursor,
+    struct wlr_surface **new_surface,
+    double *cursor_sx,
+    double *cursor_sy)
+{
+    struct kiwmi_desktop *desktop = &cursor->server->desktop;
+    struct wlr_seat *seat         = cursor->server->input.seat->seat;
+
+    double ox                     = 0;
+    double oy                     = 0;
+    struct wlr_output *wlr_output = wlr_output_layout_output_at(
+        desktop->output_layout, cursor->cursor->x, cursor->cursor->y);
+
+    wlr_output_layout_output_coords(
+        desktop->output_layout, wlr_output, &ox, &oy);
+
+    struct kiwmi_output *output = wlr_output->data;
+
+    struct wlr_surface *surface = NULL;
+    double sx;
+    double sy;
+
+    struct kiwmi_layer *layer = layer_at(
+        &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY],
+        &surface,
+        cursor->cursor->x,
+        cursor->cursor->y,
+        &sx,
+        &sy);
+
+    if (!layer) {
+        struct kiwmi_view *view = view_at(
+            desktop, cursor->cursor->x, cursor->cursor->y, &surface, &sx, &sy);
+
+        if (!view) {
+            wlr_xcursor_manager_set_cursor_image(
+                cursor->xcursor_manager, "left_ptr", cursor->cursor);
+        }
+    }
+
+    if (surface) {
+        wlr_seat_pointer_notify_enter(seat, surface, sx, sy);
```

Comment:
Should there be some more checks? `process_cursor_motion` didn’t care anymore, but it might be different the other times `cursor_refresh_focus` is called.